From 5bcedeec96fa336f3af8acb76d66ca49c82de4a4 Mon Sep 17 00:00:00 2001
From: Marcos Marado <mmarado@cyngn.com>
Date: Tue, 8 Sep 2015 15:42:28 +0100
Subject: [PATCH 1/4] Don't throw a NPE when there's no volume

If updatePublicVolumeState is called with a null volume, return instead of
blowing up with a NullPointerException.

Change-Id: I3d8c266958fd48fea385b111b4836d7fd1fe4a8e
---
 services/core/java/com/android/server/MountService.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/core/java/com/android/server/MountService.java b/services/core/java/com/android/server/MountService.java
index dcb13c3d19d..89e2daf6527 100755
--- a/services/core/java/com/android/server/MountService.java
+++ b/services/core/java/com/android/server/MountService.java
@@ -749,6 +749,10 @@ class MountService extends IMountService.Stub
     }
 
     private void updatePublicVolumeState(StorageVolume volume, String state) {
+        if (volume == null) {
+            Slog.e(TAG, "failed to updatePublicVolumeState, since given volume is null");
+            return;
+        }
         final String path = volume.getPath();
         final String oldState;
         synchronized (mVolumesLock) {
-- 
2.17.1

